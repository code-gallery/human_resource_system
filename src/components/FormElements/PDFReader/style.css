.custom-document-view {
  width: 100%
}
.custom-document-view .ReactPDF__Page canvas {
  width: 100% !important;
}
.pager {
  text-align: center
}
