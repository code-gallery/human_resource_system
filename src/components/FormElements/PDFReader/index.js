import React from 'react'
import PropTypes from 'prop-types'
import { Document, Page } from 'react-pdf'
import './style.css'

export default class MyPdfViewer extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      page: null
    }
  }

  onDocumentComplete = () => {
    this.setState({ page: 1 })
  }

  render() {
    const { transcript } = this.props
    return (
      <div>
        <Document
          file={transcript}
          onLoadSuccess={this.onDocumentComplete}
          className="custom-document-view"
        >
          <Page pageNumber={this.state.page} />
        </Document>
      </div>
    )
  }
}

MyPdfViewer.propTypes = {
  transcript: PropTypes.string
}
