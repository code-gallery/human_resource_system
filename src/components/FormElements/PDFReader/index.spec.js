import React from 'react'
import PDFReader from './index'
import { shallow } from 'enzyme'

describe('<PDFReader />', () => {
  const props = { transcript: 'transcript' }
  describe('when props is valid', () => {
    it('renders correctly', () => {
      const tree = shallow(
        <PDFReader {...props} />
      )
      expect(tree).toMatchSnapshot()
    })
  })
})
