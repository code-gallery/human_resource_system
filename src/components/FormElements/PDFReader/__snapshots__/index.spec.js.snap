// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PDFReader /> when props is valid renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Document
        className="custom-document-view"
        file="transcript"
        onLoadSuccess={[Function]}
    >
        <Page
            pageNumber={null}
        />
    </Document>
</div>,
  "nodes": Array [
    <div>
      <Document
            className="custom-document-view"
            file="transcript"
            onLoadSuccess={[Function]}
      >
            <Page
                  pageNumber={null}
            />
      </Document>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <MyPdfViewer
        transcript="transcript"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": MyPdfViewer {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onDocumentComplete": [Function],
        "props": Object {
          "transcript": "transcript",
        },
        "refs": Object {},
        "state": Object {
          "page": null,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Document
                    className="custom-document-view"
                    file="transcript"
                    onLoadSuccess={[Function]}
          >
                    <Page
                              pageNumber={null}
                    />
          </Document>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <Document
                    className="custom-document-view"
                    file="transcript"
                    onLoadSuccess={[Function]}
          >
                    <Page
                              pageNumber={null}
                    />
          </Document>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <MyPdfViewer
    transcript="transcript"
/>,
}
`;
